Add comprehensive test cases for StreamAttendee and related entities

Heuy-kt opened the pull request:
This pull request introduces a new `StreamAttendeeTest` class and comprehensive test cases to ensure the proper functionality of the `StreamAttendee` entity and its related attributes.  

#### Summary of Changes  
1. **Created `StreamAttendeeTest` class.**  
2. **Added tests for basic `StreamAttendee` creation:**  
   - Verified the creation of an empty `StreamAttendee` instance.  
   - Confirmed that a null `StreamAttendee` instance is correctly handled.  

3. **Added tests for `StreamAttendee` ID scenarios:**  
   - Verified the creation of `StreamAttendee` without an ID.  
   - Verified the creation of `StreamAttendee` with an ID.  
   - Ensured `StreamAttendee` ID is not null.  
   - Ensured `StreamAttendee` ID is null.  
   - Ensured `StreamAttendee` IDs are equal.  
   - Ensured `StreamAttendee` IDs are not equal.  

4. **Added tests for `FleenStream` relationships:**  
   - Ensured `FleenStream` is not null.  
   - Ensured `FleenStream` is null.  
   - Ensured `FleenStreams` are equal.  
   - Ensured `FleenStreams` are not equal.  

5. **Added tests for `Member` relationships:**  
   - Ensured `Member` is not null.  
   - Ensured `Member` is null.  
   - Ensured `Members` are equal.  
   - Ensured `Members` are not equal.  

6. **Added tests for `StreamAttendee` properties:**  
   - Ensured request-to-join status is not null.  
   - Ensured request-to-join status is null.  
   - Ensured `isAttending` is not null.  
   - Ensured `isAttending` is null.  
   - Ensured attendee comment is not null.  
   - Ensured attendee comment is null.  
   - Ensured organizer comment is not null.  
   - Ensured organizer comment is null.  

## Why This Change Is Necessary  
These tests enhance the reliability of the codebase by verifying the behavior of the `StreamAttendee` entity under various conditions. They ensure relationships, properties, and logic function as intended.  

## Additional Notes  
- All tests were written using JUnit and follow proper assertions for validation.  
- Future modifications to the `StreamAttendee` entity or its relationships can be easily validated against these tests.

aalamu commented:
Overall, we've a good test suites. We can improve by adding test cases for the StreamAttendee method including 

```
- markRequestAsPending() and then confirm the getRequestToJoinStatus() is ```PENDING```
- markAsNotAttending() and then confirm isAttending() is ```false```
- isRequestToJoinApproved() and then confirm if it is approved.
- isRequestToJoinDisapproved() and then confirm if it is disapproved
- And the other methods
```

Heuy-kt commented:
# Pull Request: Add Unit Tests for Attendance and Request-to-Join Status

### Description
This pull request adds unit tests to verify various scenarios related to attendance (`isAttending`) and the request-to-join status in the application. The tests ensure the correct behavior of the system in different states, enhancing the reliability and robustness of the application.

### Changes Included
1. **Unit Test for Attendance:**
   - Added a test to verify `isAttending` was false.
   - Added a test to verify `isAttending` was true.

2. **Unit Test for Request-to-Join Status:**
   - Added a test to verify the status was `PENDING`.
   - Added a test to verify the status was `APPROVED`.
   - Added a test to verify the status was `DISAPPROVED`.

### Purpose
- Ensure the application handles attendance and request-to-join statuses correctly in various scenarios.
- Improve the test coverage and maintainability of the codebase.
- Identify and prevent potential regressions in these specific functionalities.
